games/static/games/proto.js

```diff
@@ -204,18 +199,32 @@ class Board {
         }
     }
 
+    determineDirection() {
+        const aliensAtLeftBorder = this.squares[this.alienLocations[0]].isLeftBorder()
```

Comment:
sounds like AreAliensAtLeftBorder could be method on board class
also i would name/explain what is this square:
this.squares[this.alienLocations[0]]

and what is this one:
this.squares[this.alienLocations[this.alienLocations.length - 1]]